### PR TITLE
feat: add with-reflexjs example

### PR DESCRIPTION
**Pull request by shadcn**
This PR adds an example for using Next.js with [Reflexjs](https://reflexjs.org).

Thank You

## Documentation / Examples

- [x] Make sure the linting passes

### CODE REVIEW

File: examples/with-reflexjs/.babelrc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "presets": [
+    [
+      "next/babel",
+      {
+        "preset-react": {
+          "runtime": "automatic",
+          "importSource": "reflexjs"
```
Comment (leerob) [inline examples/with-reflexjs/.babelrc:8]: Why is this needed?

File: examples/with-reflexjs/pages/_app.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import * as React from 'react'
```
Comment (leerob) [inline examples/with-reflexjs/pages/_app.tsx:1]: This shouldn't be needed 👍

File: examples/with-reflexjs/src/theme.ts [hunk 3]
```diff
@@ -0,0 +1,817 @@
...
+        color: 'prism.boolean',
+      },
+      '.property': {
+        color: 'prism.property',
+      },
+      '.namespace': {
+        color: 'prism.namespace',
+      },
+      '.highlight': {
+        bg: 'prism.highlight',
```
Comment (leerob) [inline examples/with-reflexjs/src/theme.ts:451]: Could we use a smaller theme?

File: examples/with-reflexjs/pages/_app.tsx (hunk 2 above)
Comment (arshad) [reply examples/with-reflexjs/pages/_app.tsx:1]: Thanks. Removed.

File: examples/with-reflexjs/src/theme.ts (hunk 3 above)
Comment (arshad) [reply examples/with-reflexjs/src/theme.ts]: Sure. I pushed a smaller one.

File: examples/with-reflexjs/.babelrc (hunk 1 above)
Comment (arshad) [reply examples/with-reflexjs/.babelrc:8]: This sets a custom pragma to convert style props to sx props. So that you don't need to import it for each individual files.

https://reflexjs.org/docs/installation/nextjs#add-the-babel-preset

**Review by leerob — APPROVED** [review]
Thank you!
